doc/developer/cmakeutilities.md

```diff
@@ -0,0 +1,588 @@
+\page cmakeutilities Cmake utilities for developing using Radium 
+[TOC]
+
+The Radium environment could be used in several ways 
+ 1. developing some specific applications, 
+ 2. extending the Radium capabilities by developing new libraries
+ 3. integrating new functionalities in the Radium applications using plugins.
+
+As Radium rely on the [cmake](https://cmake.org/documentation/) build and configuration system, this 
+documentation focus on how to configure your build system to use and extend Radium.
+
+# Radium cmake utilities
+
+Once installed, either from source or from pre-built binaries, Radium could be used and extended in any application, library or plugin as a cmake package.
+
+To integrate the Radium libraries into your build-chain configuration, you have to ask cmake to find the Radium package.
+As for any cmake package, this could be done by adding the following line into your `CMakeLists.txt` file:
+~~~{.cmake}
+find_package(Radium REQUIRED Core Engine)
+~~~
+Remember to configure your project by giving the cmake command line option `-DRadium_DIR=/path/to/installed/Radium/lib/cmake/Radium` so that the `find_package` command could find the Radium package definition.
+Once found, the Radium package defines a cmake target for each requested components (`Core`and `Engine` in the example above) in the cmake namespace `Radium`
+
+When configuring your own target in your `CMakeLists.txt` file, you just need to link with those targets to get access to all the public interface of Radium (include search path, libraries, ...).
+
+The radium package also defines several cmake functions, described below, that you can use to ease the configuration of your application, library or plugin, mainly to install them in a relocatable way while allowing their use from their own build-tree. 
+
+The functions defined by the Radium package are the following:
+1. Client application configuration
+  - [configure_radium_app](#configure_radium_app)
+2. Extending Radium through libraries
+  - [configure_radium_library](#configure_radium_library).
+  - [installTargetResources](#installTargetResources).
+  - [configure_radium_package](#configure_radium_package)
+  - [radium_exported_resources](#radium_exported_resources) 
+3. Extending Radium through plugins 
+  - [configure_radium_plugin](#configure_radium_plugin).
+
+## Client application configuration
+### Function configure_radium_app {#configure_radium_app}
+~~~{.cmake}
+configure_radium_app( 
+    NAME applicationName                        # The name of the executable target to configure as a relocatable application
+    [USE_PLUGINS]                               # Set this if the application uses installed plugins. 
+    [RESOURCES ResourceDir1 ResourceDir2 ...]   # List of directories to install as application resources
+)
+~~~
+
+This function configure the executable target `<NAME>` for installation so that the installation directory is relocatable and distributable.
```

Comment:
```suggestion
This function configures the executable target `<NAME>` for installation so that the installation directory is relocatable and distributable.
```